### PR TITLE
expose canvas to options, so I can save the image. fix broken mouseen…

### PR DESCRIPTION
**Pull request by w3sami**
…ter/leave. named functions

Sorry for the whole file being modified, basically the intendation does it :/
using named function is just good practise, and makes for cleaner code :)

### CODE REVIEW

**Comment by pwambach** [comment]
Hey,

thx for the PR. I'm sorry but i cannot merge it because you edited the build file instead of the source files in the src folder. I can barely see the changes you made.
Could you please edit the source files and make several commits and PRs for each of your points above with a description what you have done and what was broken.

I don't see any advantage for naming a directives function besides debugging and probably reuse (not in this case). If there isn't another one (please tell me if there is :) ) i prefer to keep the function inline but to give it a name like:

```
.directive('pwCanvas', function pwCanvasPostLink(){...});
```

It's more a matter of personal preferences.

Cheers phil

**Comment by w3sami** [comment]
Hi,

Yeah sorry did it in a hurry. I will post smaller prs tomorrow. And on the
subject of named functions, it of course is a matter of taste. Imo it does
have a few key advantages, one u mentioned, debugging, other bigger one is
the parse order. Named get parsed to function scope at 1st pass and are
thereafter available through out. So your "initListeners" function can be before any function declaration. Ordering the code
that way makes it more readable. Imo, of course, well my peers here too ;)
in short,  it eliminates init order issues

Sami

**Comment by w3sami** [comment]
And, it made the whole thing one step less indented, which is always a good thing. Anyway, didn't include these in the new pr's for now 

ps. there's an extra space in var getOffset line 126-141, fyi
